src/components/feed/actions/index.tsx

```diff
@@ -0,0 +1,139 @@
+// Components
+import ActionTriggersFunction from "@components/feed/actions/triggers-function";
+import ActionGoesToPage from "@components/feed/actions/goes-to-page";
+import ActionOpensSite from "@components/feed/actions/opens-site";
+
+// Types
+import { Attachment } from "@utils/types/feed/post";
+import {
+  ActionOpensLink,
+  ActionTriggersFunction as ActionTriggersFunctionType,
+} from "@utils/types/feed/action";
+import MaterialIcon from "@components/global/icon";
+
+/**
+ * Decides what type of action to render
+ * @param action An action object
+ */
+const PostAction = ({
+  action,
+}: {
+  action: ActionTriggersFunctionType | ActionOpensLink;
+}): JSX.Element => {
+  switch (action.type) {
+    case "function":
+      return <ActionTriggersFunction action={action} />;
+    case "link":
+      switch (action.external) {
+        case true:
+          return <ActionOpensSite action={action} />;
+        default:
+          return <ActionGoesToPage action={action} />;
+      }
+    default:
+      return <div className="btn">Invalid</div>;
+  }
+};
+
+/**
+ * Displays actions according to attachments
+ * @param attachments An array of attachments
+ */
+const PostActions = ({
+  attachments,
+}: {
+  attachments: Array<Attachment>;
+}): JSX.Element => {
+  // Initiate an empty array of actions
+  let actions: Array<ActionTriggersFunctionType | ActionOpensLink> = [];
```

Comment:
This one could be just a map function
```
let actions = attachments.map((attachment) => {
    switch (attachment.type) {
   ... 
    }
})
```
This way it would be faster in terms of performance than a for loop
and it would be a bit more read able than `for i of x`

Reply:
I feel like using a map function would be way too complicated for what it's worth. I will rewrite it to use `for i of x` if you think speed is an issue.